### PR TITLE
ISAICP-6381: Show the topics with hierarchy.

### PR DESCRIPTION
**Pull request by idimopoulos**

### CODE REVIEW

**Review by claudiu-cristea — CHANGES_REQUESTED** [review]
I had to reject because this looks very bad:

Screenshot 2021-05-25 at 22 58 06

File: config/sync/search_api.index.published.yml [hunk 1]
```diff
@@ -328,6 +328,7 @@ field_settings:
     dependencies:
       module:
         - meta_entity
+        - meta_entity
```
[the remark this answers is not in the document]
Comment (idimopoulos) [reply config/sync/search_api.index.published.yml]: I saw today that I had a newer version of the facets installed on my machine which was causing some strange behavior. I think this was the culprit. Now it shows fine for me too.

File: tests/features/joinup_search/search.feature [hunk 2]
```diff
@@ -37,11 +37,11 @@ Feature: Global search
...
-      | Any topic                   |
-      | Demography (2)              |
-      | Social and Political (2)    |
-      | Info (1)                    |
-      | Statistics and Analysis (1) |
+      | Any topic                     |
+      | Social and Political (2)      |
+      | • Demography (2)              |
+      | Info (1)                      |
+      | • Statistics and Analysis (1) |
```
Comment (claudiu-cristea) [inline tests/features/joinup_search/search.feature:44]: 1. If I'm not missing something, the following statement is true:
   > The sum of child term counts equals the parent term count.
   
   But we need a regression test for this. Preferable to test more than one parent with many child terms.
1. We need also to add a test that asserts search result on 2-3 child terms, then the search results when filtering on parent. The test should prove that all the search entries of child terms are the same as results on parent term. Order is also important.

**Review by claudiu-cristea — CHANGES_REQUESTED** [review]
I'm requesting to extend the test coverage.

**Pull request merged**